jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/DMatrix.java

```diff
@@ -197,6 +207,17 @@ public DMatrix(float[] data, int nrow, int ncol, float missing) throws XGBoostEr
     handle = out[0];
   }
 
+  /**
+   * create DMatrix from dense matrix
+   * @param matrix instance of BigDenseMatrix
+   * @param missing the specified value to represent the missing value
+   */
+  public DMatrix(BigDenseMatrix matrix, float missing) throws XGBoostError {
+    long[] out = new long[1];
+    XGBoostJNI.checkCall(XGBoostJNI.XGDMatrixCreateFromMatRef(matrix.address, (int) matrix.nrow, (int) matrix.ncol, missing, out));
```

Comment:
Looks really good @michalkurka ! One thing worth considering, though, would be to either change the nrow/ncol type in `BigDenseMatrix` to `int` or `XGDMatrixCreateFromMatRef` arguments from `int` to `long` (and add a validation to prevent a potential overflow I mentioned in the other comment).

Imho no point in having `long` in `BDM` if we'll always cast it to `int` anyway.

The question is: whether we want to be compatible with other constructors (which take `int`) or do we want to be able to handle an obscure edge case where `nrow > Integer.MAX_VALUE` and `nrows * ncols * FLOAT_BYTE_SIZE < Long.MAX_VALUE`.

Reply:
I think we should stay just with ints for now. Currently you cannot specify a response that has > Integer.MAX_VALUE values. So having longs here as no real world benefit

Reply:
@michalkurka yes and ints would be consistent with the rest of the DMatrix API 👍